src/arrayMethodJoin.js

```diff
@@ -5,8 +5,39 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let newSeparator = separator;
+
+    if (this === undefined || this.length === 0) {
+      return result;
+    }
+
+    if (separator === undefined) {
+      newSeparator = ',';
+    }
+
+    if (separator === null) {
+      newSeparator = 'null';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
```

Comment:
```suggestion
    for (let i = 0; i < this.length; i++) {
```
Consider looping for every element and check in the loop: if the index is the last element - don't add the separator